src/Reflection/ReflectionIntersectionType.php

```diff
@@ -25,12 +24,12 @@ public function __construct(
         IntersectionType $type,
     ) {
         /** @var non-empty-list<ReflectionNamedType> $types */
```

Comment:
In theory not needed? Should be inferred :thinking:

Reply:
```
ERROR: MixedPropertyTypeCoercion - src/Reflection/ReflectionIntersectionType.php:33:24 - $this->types expects 'non-empty-list<Roave\BetterReflection\Reflection\ReflectionNamedType>',  parent type `array<array-key, Roave\BetterReflection\Reflection\ReflectionNamedType>` provided (see https://psalm.dev/196)
        $this->types = $types;
```

Reply:
We just did an `array_map()` with a closure with well defined return types, and it failed inference? :O